decompiler/pipeline/controlflowanalysis/variable_name_generation.py

```diff
@@ -102,33 +71,40 @@ class HungarianScheme(RenamingScheme):
         Float: {16: "h", 32: "f", 64: "d", 80: "ld", 128: "q", 256: "o"},
         Integer: {8: "ch", 16: "s", 32: "i", 64: "l", 128: "i128"},
     }
-
-    custom_var_names = {
-        "tmp_": "Tmp",
-        "loop_break": "LoopBreak"
-    }
+    
 
     def __init__(self, task: DecompilerTask) -> None:
         super().__init__(task)
         self._name = VariableNameGeneration.name
-        self._var_name: str = task.options.getstring(f"{self._name}.variable_name", fallback="Var")
         self._pointer_base: bool = task.options.getboolean(f"{self._name}.pointer_base", fallback=True)
         self._type_separator: str = task.options.getstring(f"{self._name}.type_separator", fallback="")
         self._counter_separator: str = task.options.getstring(f"{self._name}.counter_separator", fallback="")
-    
-
-    def renameVariableNames(self):
-        """Rename all collected variables to the hungarian notation."""
-        for var in self._variables:
-            if self.alread_renamed(var._name):
-                continue
-            counter = _get_var_counter(var.name)
-            var._name = self._hungarian_notation(var, counter if counter else "")
+        self._variable_counter: dict[Variable, Integer] = {}
+        self._counter: Integer = 0
             
 
-    def _hungarian_notation(self, var: Variable, counter: int) -> str:
-        """Return hungarian notation to a given variable."""
-        return f"{self._hungarian_prefix(var.type)}{self._type_separator}{self.custom_var_names.get(var._name.rstrip(counter), self._var_name)}{self._counter_separator}{counter}"
+    def _get_counter(self, var: Variable) -> Integer:
+        """Look up if variable already has a counter, if not assign new one"""
+        if var not in self._variable_counter: # If the set _really_ works, then it's enough to just return a increasing number
+            self._variable_counter[var] = self._counter
+            self._counter += 1      
+        return self._variable_counter[var]
+
+
+    def _get_name_identifier(self, name: str) -> str:
+        """Return identifier by purging non alpha chars + capitalize the string. If string is too short, return generic"""
+        if len(name) < 2:
+            return "Var"
+        return "".join(filter(str.isalpha, name)).capitalize()
+
+
+    def getVariableName(self, var: Variable) -> str:
+        """Return hungarian notation to a given variable. If no prefix exists, make the first char lower case."""
+        newName = self._get_name_identifier(var._name)
+        if (prefix := self._hungarian_prefix(var.type)):
+            return f"{prefix}{self._type_separator}{newName}{self._counter_separator}{self._get_counter(var.name)}"
+        newName = newName[0].lower() + newName[1:]
```

Comment:
Why not return the uncapitalized name in function `_get_name_identifier` and capitalize the name in line 105?
Then you do not need line 106.

Reply:
Done, but will make with the next conversation be ugly